### PR TITLE
Branch 1

### LINKED ISSUE #3: how to reference a PR to an issue

### PR DESCRIPTION
**Pull request by Amirsharifico**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const express = require('express')
+const app= express();
+
+app.get('/', (req, res) =>{
+    res.send('helooooooooo')
+})
+
+app.listen(3000, () => console.log('Listening tom port 3000...'))
```
Comment (saberistic) [inline index.js:8]: typo in log

Comment (Amirsharifico) [reply index.js:8]: typo fixed

File: package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "express-rest-api",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
```
Comment (saberistic) [inline package.json:6]: needs a start script to know how to start this server

**Review by saberistic — CHANGES_REQUESTED** [review]
Few comments added

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+bower_components
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
```
[the remark this answers is not in the document]
Comment (Amirsharifico) [reply .gitignore:46]: I just used (https://www.toptal.com/developers/gitignore) and copied all, but sure, will do it more precisely

[the remark this answers is not in the document]
Comment (Amirsharifico) [reply .gitignore:46]: **this is a test**

[the remark this answers is not in the document]
Comment (Amirsharifico) [reply .gitignore:46]: ` this is a test`

[the remark this answers is not in the document]
Comment (saberistic) [reply .gitignore:46]: yeah not a fan of these ready to go tools. I rather keep things up to point of what is really there and need to be ignored

File: package.json (hunk 2 above)
Comment (Amirsharifico) [reply package.json:6]: I installed nodemon , we also had server-static so which one i should use:
`"serve": "static-server src -p 3000"`
or
` "start": "nodemon index.js"`
?

Comment (saberistic) [reply package.json:6]: nodemon, static-serve is for serving static content, like html and js that loads in browser. This is a backend so not a static resource

File: index.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ const express = require('express')
 const app= express();
 
 app.get('/', (req, res) =>{
-    res.send('helooooooooo')
+    res.send('This message is from insid of express-rest-api app by A5HKIAN ')
```
Comment (saberistic) [inline index.js:5]: typo

File: package.json [hunk 5]
```diff
@@ -18,6 +19,7 @@
   },
   "homepage": "https://github.com/Amirsharifico/express-rest-api#readme",
   "dependencies": {
-    "express": "^4.18.2"
+    "express": "^4.18.2",
+    "static-server": "^2.2.1"
```
Comment (saberistic) [inline package.json:23]: should this be nodemon instead of static-serve?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,21 @@
+
+FROM node:18-alpine
+
+
+RUN npm install -g nodemon
```
Comment (saberistic) [inline Dockerfile:5]: if we add nodemon as dependency in package.json, then we don't need to install it globally here

**Pull request merged**